helios-core/src/main/kotlin/helios/instances/instances.kt

```diff
@@ -59,7 +69,9 @@ fun Byte.Companion.encoder() = object : Encoder<Byte> {
 
 fun Byte.Companion.decoder() = object : Decoder<Byte> {
   override fun decode(value: Json): Either<DecodingError, Byte> =
-    value.asJsNumber().flatMap { it.toByte() }.toEither { NumberDecodingError(value) }
+    value.asJsNumber().toEither { JsNumberDecodingError(value) }.flatMap {
+      it.toByte().toEither { ExceptionOnDecoding(value, "Number cannot be decoded into Byte") }
```

Comment:
Are those (`toInt`, `toShort` and `toByte`) already returning a `Try`?

Reply:
Those are safe methods returning an Option if the conversion fails due to an out of range number